src/ORM/Behavior/EnumBehavior.php

```diff
@@ -0,0 +1,151 @@
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\ORM\Behavior;
+
+use ArrayObject;
+use BackedEnum;
+use Cake\Datasource\EntityInterface;
+use Cake\Event\EventInterface;
+use Cake\ORM\Behavior;
+use Cake\ORM\Entity;
+use Cake\ORM\Query;
+use ValueError;
+
+/**
+ * Enum behavior
+ *
+ * Allows you to map a specific field in your model to a backed enum
+ *
+ *  * Add behavior + config to connect a field with a backed enum
+ * ```
+ * $this->addBehavior('Enum', [
+ *     'fieldMap' => [
+ *         'published' => ArticleStatus::class
+ *     ]
+ * ]);
+ * ```
+ */
+class EnumBehavior extends Behavior
+{
+    /**
+     * Default config
+     *
+     * These are merged with user-provided configuration when the behavior is used.
+     *
+     * @var array<string, mixed>
+     */
+    protected array $_defaultConfig = [
+        'fieldMap' => [],
+    ];
+
+    /**
+     * Initialize hook
+     *
+     * @param array $config The config for this behavior.
+     * @return void
+     */
+    public function initialize(array $config): void
+    {
+        $schema = $this->_table->getSchema();
+        foreach (array_keys($this->getConfig('fieldMap')) as $field) {
+            $schema->setColumnType($field, 'enum');
+        }
+        $this->_table->setSchema($schema);
+    }
+
+    /**
+     * Transform scalar values to enum instances
+     *
+     * @param \Cake\Event\EventInterface $event The beforeMarshal event that was fired
+     * @param \Cake\Datasource\EntityInterface $entity Entity created after marshaling
+     * @param \ArrayObject $data The original request data
+     * @param \ArrayObject $options Options passed to the event
+     * @return void
+     */
+    public function afterMarshal(
+        EventInterface $event,
+        EntityInterface $entity,
+        ArrayObject $data,
+        ArrayObject $options
+    ): void {
+        $fieldMap = $this->getConfig('fieldMap');
+        /** @var \BackedEnum $enumType */
+        foreach ($fieldMap as $field => $enumType) {
+            if ($data->offsetExists($field)) {
+                $value = $data->offsetGet($field);
+                if (is_int($value) || is_string($value)) {
+                    try {
+                        $enumValue = $enumType::from($value);
+                        $entity->set($field, $enumValue);
+                    } catch (ValueError $error) {
+                        $entity->setError($field, __('Given value is not valid'));
```

Comment:
```suggestion
                        $entity->setError($field, __('cake', 'Given value is not valid'));
```
We would need to allow users to customize this message.

Reply:
I resolved this with the correct `__d('cake', '')` method